### DISCUSSION TITLE
Lesson 6: fund_and_withdraw.py script error....

### DISCUSSION THREAD

**Discussion by toddabraham** [smartcontractkit/full-blockchain-solidity-course-py#58] [Q&A] [answered]
Trying to run my fund_and_withdraw.py sript from brownie and getting the error **VirtualMachineError**: revert message.

I seem unable to find any information that helps me understand this error and how to resolve it. Is anyone able to offer guidance?

![image](https://user-images.githubusercontent.com/26919212/133978768-df26601a-6bcd-4592-a065-ccb2b55aa990.png)

**Comment by toddabraham** [comment, marked as answer]
Patrick Edit: The fix was to go back and find a typo or missing "something"


Now I'm getting this error.....
![image](https://user-images.githubusercontent.com/26919212/134005389-b6bfb9bb-e610-4554-b919-83b76aafff39.png)

**Reply by toddabraham** [reply]
I recently tried rebuilding my system after having issues with my VSCode SSH connection continually disconnecting.  I wonder if in my rebuild process I forgot to add in some important dependencies.

**Reply by PatrickAlphaC** [reply]
Thanks for making this. How did you solve the last issue? 

Can you please add:
1. Exactly what your script looks like
2. What your contract looks like
3. The text of your error

Instead of adding images?

**Reply by AlbertoSinigaglia** [reply]
you are probably passing a parameter with default value to `None` to the smart contract

**Reply by toddabraham** [reply]
@PatrickAlphaC  @AlbertoSinigaglia  

1. The VitrualMachineError: revert issue turned into ValueError: No data was returned - the call likely reverted after restarting ganache-cli.

2. SCRIPT  fund_and_withdraw.py

from brownie import FundMe
from scripts.helpful_scripts import get_account


def fund():
    fund_me = FundMe[-1]
    account = get_account()
    entrance_fee = fund_me.getEntranceFee()
    print(entrance_fee)
    print(f"The entrance fee is {entrance_fee}")
    print("Funding")
    fund_me.fund({"from": account, "value": entrance_fee})


def main():
    fund()

3. CONTRACT FundMe.sol

// SPDX-License-Identifier: MIT

pragma solidity ^0.6.6;

import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol";

contract FundMe {
    using SafeMathChainlink for uint256;

    mapping(address => uint256) public addressToAmountFunded;
    address[] public funders;
    address public owner;

    constructor() public {
        owner = msg.sender;
    }

    function fund() public payable {
        uint256 minimumUSD = 50 * 10**18;
        require(
            getConversionRate(msg.value) >= minimumUSD,
            "You need to spend more ETH!"
        );
        addressToAmountFunded[msg.sender] += msg.value;
        funders.push(msg.sender);
    }

    function getVersion() public view returns (uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(
            0x8A753747A1Fa494EC906cE90E9f37563A8AF630e
        );
        return priceFeed.version();
    }

    function getPrice() public view returns (uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(
            0x8A753747A1Fa494EC906cE90E9f37563A8AF630e
        );
        (, int256 answer, , , ) = priceFeed.latestRoundData();
        return uint256(answer * 10000000000);
    }

    // 1000000000
    function getConversionRate(uint256 ethAmount)
        public
        view
        returns (uint256)
    {
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
        return ethAmountInUsd;
    }

    modifier onlyOwner() {
        require(msg.sender == owner);
        _;
    }

    function withdraw() public payable onlyOwner {
        msg.sender.transfer(address(this).balance);

        for (
            uint256 funderIndex = 0;
            funderIndex < funders.length;
            funderIndex++
        ) {
            address funder = funders[funderIndex];
            addressToAmountFunded[funder] = 0;
        }
        funders = new address[](0);
    }
}

4. CONFIG brownie-config.yaml

dependencies:
  # - <organization/repo>@<version> 
  - smartcontractkit/chainlink-brownie-contracts@1.1.1
compiler:
  solc:
    remappings:
      - '@chainlink=smartcontractkit/chainlink-brownie-contracts@1.1.1'
dotenv: .env
networks:
  default: development
  rinkeby:
    eth_usd_price_feed: '0x594B45Fc8Fe429BfdBbC809099B77870B747205d'
    verify: True
  development:
    verify: False
  ganache-local:
    verify: False
wallets:
  from_key: ${PRIVATE_KEY}

5. ERROR TEXT

unning 'scripts/fund_and_withdraw.py::main'...
  File "brownie/_cli/run.py", line 49, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/fund_and_withdraw.py", line 22, in main
    fund()
  File "./scripts/fund_and_withdraw.py", line 8, in fund
    entrance_fee = fund_me.getEntranceFee()
  File "brownie/network/multicall.py", line 115, in _proxy_call
    result = ContractCall.__call__(*args, **kwargs)  # type: ignore
  File "brownie/network/contract.py", line 1729, in __call__
    return self.call(*args, block_identifier=block_identifier)
  File "brownie/network/contract.py", line 1539, in call
    raise ValueError("No data was returned - the call likely reverted")
ValueError: No data was returned - the call likely reverted

**Reply by toddabraham** [reply]
Backed up to the beginning of lesson 6, made a couple of changes (one was that I found my rinkeby address was incorrect).  Fixed and moving on.

**Reply by AlbertoSinigaglia** [reply]
@toddabraham  good job, just as a note, you can use three back ticks to insert code and be highlighted: 
````python
```python
from Y import X
```
````

**Comment by adele013** [comment]
Hey sorry but i dont know how you fixed the revert problem. I am having the same issue.

**Reply by cromewar** [reply]
Hello @adele013 could you please share more info about your error, the author of this thread said that his rinkeby address was incorrect, does this make any sense to you?
